### PR TITLE
adding ssmax as per the paper

### PR DESCRIPTION
**Pull request by adi-kmt**
Problems with softmax according to[ the paper](https://arxiv.org/abs/2501.19399):- 
- As context size increases, the denominator of the softmax increases (sum of exponentials of input elements), while the numerator (exponential of a single element) remains constant. This causes the attention distribution to become increasingly flat.
- Thus, Transformers trained on shorter context sizes often struggle to generalize to longer contexts due to attention fading and the computational burden of processing long sequences.
- Also, Softmax-based models struggle to retrieve key information (e.g., a specific number embedded in a long context) because attention scores are diluted across many tokens.

### CODE REVIEW

**Comment by IsNoobgrammer** [comment]
LGTM !! 

Rest I can iterate over it
